service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/authentication/ServiceClientBearerTokenCredentialProvider.java

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.service.digitaltwin.authentication;
+
+import com.microsoft.rest.credentials.ServiceClientCredentials;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+
+/**
+ * Implementation of {@link ServiceClientCredentials} that provides RBAC based bearer tokens.
+ */
+@AllArgsConstructor
+public class ServiceClientBearerTokenCredentialProvider implements ServiceClientCredentials {
```

Comment:
It's worth noting that we have a class ```ServiceClientCredentialsProvider``` that looks about the same, but that uses a SasTokenProvider rather than a BearerTokenProvider to get the authentication tokens.